cmd/fscrypt/commands.go

```diff
@@ -63,6 +69,14 @@ func setupAction(c *cli.Context) error {
 	case 0:
 		// Case (1) - global setup
 		err = createGlobalConfig(c.App.Writer, actions.ConfigFileLocation)
+		if err != nil {
```

Comment:
This code here is essentially what we want to do, but it might make the code more readable to just change this all to something like:

```go
switch c.NArg() {
case 0:
	// Case (1) - global setup
	if err := createGlobalConfig(c.App.Writer, actions.ConfigFileLocation); err != nil {
		return newExitError(c, err)
	}
	if err := setupFilesystem(c.App.Writer, "/"); err != nil {
		if errors.Cause(err) != filesystem.ErrAlreadySetup {
			return newExitError(c, err)
		}
		fmt.Fprintf(c.App.Writer, "Skipping creating /.fscrypt because it already exists.\n")
	}
case 1:
	// Case (2) - filesystem setup
	if err := setupFilesystem(c.App.Writer, c.Args().Get(0)); err != nil {
		return newExitError(c, err)
	}
default:
	return expectedArgsErr(c, 1, true)
}
return nil
``` 

And avoid all the assigning to `err` stuff.

Reply:
Agreed, updated now.